utils/transactions.ts

```diff
@@ -8,3 +8,46 @@ export const getGasEstimateForTransaction = async (txArgs: any[], method: Functi
 		throw error;
 	}
 };
+
+type TxOptions = {
+	showErrorNotification: Function;
+	showProgressNotification: Function;
+	showSuccessNotification: Function;
+};
+
+export async function tx(makeTx: Function, options?: TxOptions): Promise<void> {
+	const [contract, method, args] = makeTx();
+	let hash, wait;
+	try {
+		({ hash, wait } = await contract[method](...args));
+	} catch (e) {
+		try {
+			await contract.callStatic[method](...args);
```

Comment:
@vbstreetz could you explain what is `callStatic`?

Reply:
Oh, this is a dry run of a write transaction. It's basically a hack to retrieve the revert message of a failed transaction.